packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/UpgradeRolesForm/UpgradeRolesForm.tsx

```diff
@@ -42,21 +42,22 @@ export const UpgradeRolesForm = ({
     setRadioButtons(zeroedOutRadioButtons);
   };
 
-  const nonAdmins: RoleInfo[] = roleData.filter((_role) => {
-    return (
-      _role.permission === AccessLevel.Member ||
-      _role.permission === AccessLevel.Moderator
-    );
+  const nonAdmins: MemberResult[] = roleData.filter(({ addresses }) => {
+    return !addresses.some(({ role: r }) => r === 'admin');
   });
 
-  const nonAdminNames: string[] = nonAdmins.map((_role) => {
-    const roletext = _role.permission === 'moderator' ? '(moderator)' : '';
-    const fullText = `${(_role as any)?.displayName} - ${formatAddressShort(
-      // @ts-expect-error <StrictNullChecks/>
-      _role.Address.address,
-    )} ${roletext}`;
-    return fullText;
-  });
+  const nonAdminNames: string[] = nonAdmins.map(
+    ({ profile_name, addresses }) => {
+      const roletext = addresses.some(({ role: r }) => r === 'moderator')
+        ? '(moderator)'
+        : '';
+      // TODO: we don't actually know if this is the right address?
+      const fullText = `${profile_name} - ${formatAddressShort(
+        addresses?.[0].address,
```

Comment:
if `0` index contains a different address connected to a different community, then it might break. We should choose address of the active community (if there are multiple then maybe we display them individually)

Reply:
I think the problem here is that we need to display a non-admin address for the user to associate that user with some address to display them on the members page:
<img width="733" alt="image" src="https://github.com/user-attachments/assets/8aa7d49f-49fe-4bc0-97c3-5c2731ee5ed4">

The issue is that in this case, a user can be associated with multiple addresses, so it is not obvious how to handle this members list. The current approach of this PR is just to leave it as is before.

But if we want to resolve this in a better way, how should it be handled? Maybe the profile can be linked multiple times, once for each of their addresses? Also maybe it should be handled in a followup PR since it is a bit out of the scope of the roles removal.

Reply:
Ok seems like it is part of a separate ticket I am assigned to. I will implement this functionality in a followup PR:
https://github.com/hicommonwealth/commonwealth/issues/8524